From a8bf117b12f697cb78b445fb63d077e0ba1b710b Mon Sep 17 00:00:00 2001
From: "Song, Yoong Siang" <yoong.siang.song@intel.com>
Date: Fri, 4 Sep 2020 00:04:10 +0800
Subject: [PATCH 1/5] Revert "WORKAROUND: net: stmmac: use DMA INTM 0 for TGL
 platforms"

Reverting portion of this commit as the DMA INTM 0 WA is restructured.

This reverts commit 7e10c1ce43e4212f23a48b994c184311a2c0dbec.

Signed-off-by: Song, Yoong Siang <yoong.siang.song@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c | 3 ---
 include/linux/stmmac.h                           | 1 -
 2 files changed, 4 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index c7c1b41..7b74a4f 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -541,9 +541,6 @@ static int tgl_common_data(struct pci_dev *pdev,
 	/* Maximum TX XDP queue */
 	plat->max_combined = 2;
 
-	/* WORKAROUND: TGL has to use DMA INTM 0 to avoid intermittent reset */
-	plat->dma_cfg->tgl_wa = 1;
-
 	/* TX and RX Marvell 88E2110 PHY latency (ns) */
 	plat->phy_tx_latency_10 = 6652;
 	plat->phy_tx_latency_100 = 1152;
diff --git a/include/linux/stmmac.h b/include/linux/stmmac.h
index 9abf31b..fde9726 100644
--- a/include/linux/stmmac.h
+++ b/include/linux/stmmac.h
@@ -94,7 +94,6 @@ struct stmmac_dma_cfg {
 	int mixed_burst;
 	bool aal;
 	bool multi_msi_en;
-	bool tgl_wa;
 };
 
 #define AXI_BLEN	7
-- 
2.7.4

